From c3100c901071fce190fa32914bb7db04e2ab45e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9rgio=20M=2E=20Basto?= <sergio@serjux.com>
Date: Wed, 24 Aug 2022 01:32:51 +0100
Subject: [PATCH] Use unittest.mock on Python 3 and remove python3-mock
 dependency

also
- no cover-package = rfpkg
- remove pytest coverage execution
---
 conf/zsh-completion/_rfpkg |  1 -
 setup.cfg                  |  9 +++-----
 setup.py                   | 17 +++++---------
 test/test_retire.py        | 45 +++++++++++++++++++++++++++++++++++---
 tests-requirements.txt     |  2 +-
 5 files changed, 52 insertions(+), 22 deletions(-)

diff --git a/conf/zsh-completion/_rfpkg b/conf/zsh-completion/_rfpkg
index d1cfe86..15ba1e3 100644
--- a/conf/zsh-completion/_rfpkg
+++ b/conf/zsh-completion/_rfpkg
@@ -399,7 +399,6 @@ _rfpkg() {
   _arguments -C \
     '(- :)'{-h,--help}'[show help message and exit]' \
     '(-C --config)'{-C,--config}'[specify a config file to use]:config file:_files' \
-    '--dist[override the discovered distribution]:distribution' \
     '--release[override the discovered release]:release' \
     '--user[override the discovered user name]:user' \
     '--path[define the directory to work in (defaults to cwd)]:working direcory:_directories' \
diff --git a/setup.cfg b/setup.cfg
index a38d7e6..f2746be 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -5,13 +5,10 @@ test = pytest
 formats = bztar
 
 [tool:pytest]
-cover-package = rfpkg
-addopts = --cov=rfpkg
+# additional values for 'addopts' can be:
+# --cov-report html
+# -ra -q
 testpaths = test
 
 [flake8]
 max-line-length = 100
-
-[egg_info]
-tag_build = 
-tag_date = 0
diff --git a/setup.py b/setup.py
index 8d353f3..526370c 100755
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,6 @@
 #!/usr/bin/python
 
 import os
-import sys
 
 from setuptools import find_packages, setup
 
@@ -23,20 +22,13 @@ tests_requirements = os.path.join(project_dir, 'tests-requirements.txt')
 
 with open(requirements, 'r') as f:
     install_requires = [line.strip() for line in f]
+install_requires += [
+    'distro',
+]
 
 with open(tests_requirements, 'r') as f:
     tests_require = [line.strip() for line in f]
 
-ver = sys.version_info
-if ver[0] <= 2 and ver[1] < 7:
-    tests_require += [
-        'unittest2'
-    ]
-else:
-    install_requires += [
-        'distro',
-    ]
-
 
 setup(
     name="rfpkg",
@@ -76,5 +68,8 @@ setup(
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
     ],
 )
diff --git a/test/test_retire.py b/test/test_retire.py
index 5296c1d..35b87c0 100644
--- a/test/test_retire.py
+++ b/test/test_retire.py
@@ -2,8 +2,14 @@
 
 import os
 import shutil
-import unittest
-import mock
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest
+try:
+    from unittest import mock
+except ImportError:
+    import mock
 from six.moves import configparser
 import tempfile
 import subprocess
@@ -53,7 +59,7 @@ class RetireTestCase(unittest.TestCase):
         return out.strip()
 
     def _fake_client(self, args):
-        config = configparser.SafeConfigParser()
+        config = configparser.ConfigParser()
         config.read(TEST_CONFIG)
         with mock.patch('sys.argv', new=args):
             client = rfpkgClient(config)
@@ -100,3 +106,36 @@ class RetireTestCase(unittest.TestCase):
         self.assertEqual(len(client.cmd.push.call_args_list), 1)
         self.assertEqual(PkgDB.return_value.retire_packages.call_args_list,
                          [mock.call('rfpkg', 'master', namespace='rpms')])
+
+    """
+    @mock.patch("requests.get", new=lambda *args, **kwargs: mock.Mock(status_code=404))
+    @mock.patch('rpmfusion_cert.read_user_cert')
+    @mock.patch('rfpkgdb2client.PkgDB')
+    def test_package_is_retired_already(self):
+        self._setup_repo('ssh://git@pkgs.example.com/rfpkg')
+        with open(os.path.join(self.tmpdir, 'dead.package'), 'w') as f:
+            f.write('dead package')
+
+        args = ['rfpkg', '--release=master', 'retire', 'my reason']
+        client = self._fake_client(args)
+        client.log = mock.Mock()
+        client.retire()
+        args, kwargs = client.log.warn.call_args
+        self.assertIn('dead.package found, package or module is already retired',
+                      args[0])
+
+    @mock.patch(
+        "requests.get",
+        new=lambda *args, **kwargs: mock.Mock(
+            status_code=200, ok=True, json=lambda: {"state": "archived"}
+        ),
+    )
+    def test_package_on_retired(self):
+        self._setup_repo("ssh://git@pkgs.example.com/rfpkg")
+        args = ["rfpkg", "--release=master", "retire", "my reason"]
+
+        client = self._fake_client(args)
+        client.retire()
+        args, kwargs = client.log.error.call_args
+        self.assertIn("retire operation is not allowed", args[0])
+    """
diff --git a/tests-requirements.txt b/tests-requirements.txt
index cca4322..be75121 100644
--- a/tests-requirements.txt
+++ b/tests-requirements.txt
@@ -3,6 +3,6 @@ coverage<5.0.0
 cccolutils
 gitpython
 freezegun
-rpm
+rpm < 0.0.3
 pytest
 pytest-cov
-- 
2.37.3

